onnxruntime/python/tools/transformers/models/sam2/convert_to_onnx.py

```diff
@@ -105,41 +125,36 @@ def parse_arguments():
     return args
 
 
-def main():
-    args = parse_arguments()
+def optimize_sam2_model(onnx_model_path, optimized_model_path, use_gpu: bool, float16: bool):
+    print(f"Optimizing {onnx_model_path} to {optimized_model_path} with float16={float16} and use_gpu={use_gpu}...")
 
-    checkpoints_dir = os.path.join(args.sam2_dir, "checkpoints")
-    sam2_config_dir = os.path.join(args.sam2_dir, "sam2_configs")
-    if not os.path.exists(args.sam2_dir):
-        raise FileNotFoundError(f"{args.sam2_dir} does not exist. Please specify --sam2_dir correctly.")
+    # Import from source directory.
+    transformers_dir = os.path.normpath(os.path.join(os.path.dirname(__file__), "..", ".."))
+    if transformers_dir not in sys.path:
+        sys.path.insert(0, transformers_dir)
+    from optimizer import optimize_model
 
-    if not os.path.exists(checkpoints_dir):
-        raise FileNotFoundError(f"{checkpoints_dir} does not exist. Please specify --sam2_dir correctly.")
+    optimized_model = optimize_model(onnx_model_path, model_type="sam2", opt_level=1, use_gpu=use_gpu)
+    if float16:
+        optimized_model.convert_float_to_float16(keep_io_types=False)
```

Comment:
This doesn't have to be fixed in this PR but should we export + optimize with float16 precision directly instead of using float32 precision and then calling `convert_float_to_float16` afterwards for new models such as SAM2?

Reply:
The original checkpoint is fp32. User has option to output fp32 (for testing purpose) or fp16 optimized model.